modules/usbip.nix

```
@@ -0,0 +1,58 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  usbipd-win-auto-attach = pkgs.fetchurl {
+    url = "https://raw.githubusercontent.com/dorssel/usbipd-win/9e259e74e225ed9321484366a79545b00a18185c/Usbipd/wsl-scripts/auto-attach.sh";
+    hash = "sha256-KJ0tEuY+hDJbBQtJj8nSNk17FHqdpDWTpy9/DLqUFaM=";
+  };
+
+  cfg = config.wsl.usbip;
+in
+{
+  options.wsl.usbip = with types; {
+    enable = mkEnableOption "USB/IP integration";
+    autoAttach = mkOption {
+      type = listOf str;
+      default = [ ];
+      example = [ "4-1" ];
+      description = "Auto attach devices with provided Bus IDs.";
+    };
+  };
+
+  config = mkIf (config.wsl.enable && cfg.enable) {
```

```suggestion
  config = mkIf cfg.enable {
```
or do we have similar checks on other modules?

---

Yeah, that is where I got it, I'm not sure how likely it is that someone configures the sub option but not the top one. But it could allow for some kind of composition/partial configuration. Thought it had a potential use case. But I'm fine either way